File: my_py_package/my_py_package/subscriber_node.py
#!/usr/bin/env python3

import time
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import JointState

"""
  @author: Mehmet Kahraman
  @date: 21.02.2024
  @about: Subscriber node
"""

class MyROSClass(Node):

    def __init__(self):

        super().__init__('subscriber_node')

        # publisher
        self.position = 0

        # subscriber
        self.js_subscription = self.create_subscription(JointState, "/joint_states", self.subscriber_callback, 10)
        self.js_subscription  # prevent unused variable warning

        time.sleep(1)
        self.get_logger().info("Subscriber node is ready.")


    def subscriber_callback(self, msg):
        self.position = msg.position[0]
        self.get_logger().info(' JS subscriber position received: "%s" ' %self.position)


def main(args=None):
    rclpy.init(args=args)
    node = MyROSClass()
    rclpy.spin(node)

    Node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: my_cpp_package/src/subscriber_node.cpp
#include <chrono>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include "sensor_msgs/msg/joint_state.hpp"

/**
 * @author: Mehmet Kahraman
 * @date: 03.10.2023
 * @about: Subscriber node
 **/

class MyROSClass : public rclcpp::Node
{

  private:
    
    double position_value = 0.0;

    rclcpp::Subscription<sensor_msgs::msg::JointState>::SharedPtr js_subscription;
    sensor_msgs::msg::JointState js_message;


  public:

    MyROSClass() : Node("subscriber_node")
    {

      // Subscriber
      js_subscription = this->create_subscription<sensor_msgs::msg::JointState>("/joint_states", 10, 
                        std::bind(&MyROSClass::subscriber_callback, this, std::placeholders::_1));
      auto js_message = sensor_msgs::msg::JointState();

      std::this_thread::sleep_for(std::chrono::seconds(1));

    }

    void subscriber_callback(const sensor_msgs::msg::JointState::SharedPtr msg)
    { 
      position_value = msg->position[0];
      RCLCPP_INFO(this->get_logger(), "JS subscriber position received: '%f'", position_value);
    }


};

int main(int argc, char *argv[])
{
  rclcpp::init(argc, argv);
  RCLCPP_INFO(rclcpp::get_logger("subscriber_node"), "Node initialized.");
  auto node = std::make_shared<MyROSClass>();
  rclcpp::spin(node);

  rclcpp::shutdown();
  return 0;
}